wallet/udb/upgrades.go

```diff
@@ -1026,6 +1030,77 @@ func importedXpubAccountUpgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte
 	return unifiedDBMetadata{}.putVersion(metadataBucket, newVersion)
 }
 
+// genesisPrevScripter fulfills the blockcf2.PrevScripter interface but only
+// for the genesis block which doesn't have any previous scripts.
+type genesisPrevScripter struct{}
+
+func (_ genesisPrevScripter) PrevScript(*wire.OutPoint) (uint16, []byte, bool) {
+	return 0, nil, false
+}
+
+func blockcf2Upgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte, params *chaincfg.Params) error {
+	const oldVersion = 13
+	const newVersion = 14
+
+	metadataBucket := tx.ReadWriteBucket(unifiedDBMetadata{}.rootBucketKey())
+
+	// Assert that this function is only called on version 13 databases.
+	dbVersion, err := unifiedDBMetadata{}.getVersion(metadataBucket)
+	if err != nil {
+		return err
+	}
+	if dbVersion != oldVersion {
+		return errors.E(errors.Invalid, "blockcf2Upgrade inappropriately called")
+	}
+
+	txmgrBucket := tx.ReadWriteBucket(wtxmgrBucketKey)
+
+	// Drop all existing blockcf filters by deleting then recreating the
+	// cfilter bucket.
+	//
+	// TODO: Verify if this isn't too big to do in a single tx for mainnet
+	// dbs.
```

Comment:
didn't see this mentioned in the pull request message so just commenting for visibility

Reply:
I tested it manually on my machine (linux, 64 bits) on mainnet but I'm not sure whether there are differences between archs in boltdb.

I couldn't find any docs on transaction size limits and it's not super obvious to me just by reading the boltdb code for `DeleteBucket` what are the hard limits on it. Any tips?

Reply:
Probably just need to experiment with different systems and different resource limits.